utils/io/progress.go

```diff
@@ -1,19 +1,19 @@
 package io
 
-import "io"
-
 // You may implement this interface to display progress indication of files transfer (upload / download)
-type Progress interface {
-	// Initializes a new progress indication for a new file transfer.
-	// Input: 'total' - file size, 'prefix' - optional description, 'filePath' - path of the file being transferred (for description purposes only).
+type ProgressMgr interface {
+	// Initializes a new reader progress indication for a new file transfer.
+	// Input: 'total' - file size, 'prefix' - optional description, 'extraInformation' -extra information for disply.
 	// Output: progress indication id
-	New(total int64, prefix, filePath string) (id int)
+	NewReaderProgressBar(total int64, prefix, extraInformation string) (bar ProgressBar)
 	// Replaces an indication (with the 'replaceId') when completed. Used when an additional work is done as part of the transfer.
-	NewReplacement(replaceId int, prefix, filePath string) (id int)
-	// Used to wrap an io.Reader in order to track the bytes reading count of the file transfer, and update indication 'id' accordingly.
-	ReadWithProgress(id int, reader io.Reader) io.Reader
+	AddNewReplacementSpinner(replaceId int, prefix, extraInformation string) (id int)
+	// Returns the requested progress bar
+	GetProgressBar(id int) (bar ProgressBar)
 	// Aborts a progress indication. Called on both successful and unsuccessful operations
 	Abort(id int)
 	// Quits the whole progress mechanism
 	Quit()
+	// Increases the general progress bar total count by given n.
```

Comment:
// Increases the general progress bar total count by given n. --> // Increments the general progress total count by given n.